src/gui/wizard/owncloudsetuppage.cpp

```diff
@@ -223,20 +189,17 @@ void OwncloudSetupPage::initializePage()
     // If url is overriden by theme, it's already set and
     // we just check the server type and switch to second page
     // immediately.
-    if (Theme::instance()->overrideServerUrl().isEmpty()) {
-        _ui.leUrl->setFocus();
-    } else if (!Theme::instance()->forceOverrideServerUrl()) {
+    if (!Theme::instance()->forceOverrideServerUrl()) {
         if (nextButton) {
             nextButton->setFocus();
         }
-    } else {
+
         setCommitPage(true);
         // Hack: setCommitPage() changes caption, but after an error this page could still be visible
         setButtonText(QWizard::CommitButton, tr("&Next >"));
         validatePage();
         setVisible(false);
     }
```

Comment:
Are you sure `else` part is not needed with `_ui.leUrl->setFocus();` here?

Reply:
No. I find this code really confusing. I initially thought that the code never branches into `else if (!Theme::instance()->forceOverrideServerUrl())` but it turns out that you can have `forceOverrideServerUrl()` returning `true` while `overrideServerUrl()` returns an empty string. But we don't need a call to `leUrl->setFocus()` as this widget is focused by default.